go/libraries/doltcore/doltdb/foreign_key_coll.go

```diff
@@ -61,18 +61,41 @@ func (f *ForeignKeyViolationError) Error() string {
 		}
 		key, _ := f.ViolationRows[i].NomsMapKey(f.Schema).Value(context.Background())
 		val, _ := f.ViolationRows[i].NomsMapValue(f.Schema).Value(context.Background())
-		valSlice, _ := val.(types.Tuple).AsSlice()
-		all, _ := key.(types.Tuple).Append(valSlice...)
-		str, _ := types.EncodedValue(context.Background(), all)
+
+		str := getRowValuesAsString(key, val)
 		sb.WriteRune('\n')
 		sb.WriteString(str)
 	}
+
+	var refCols []string
+	for _, tct := range f.ForeignKey.TableColumns {
+		col, _ := f.Schema.GetAllCols().GetByTag(tct)
+		refCols = append(refCols, col.Name)
+	}
+
+	fKeyErrorMsg := fmt.Sprintf("a foreign key constraint fails (CONSTRAINT `%s` FOREIGN KEY (`%s`) REFERENCES `%s` (`%s`)) on:%s", f.ForeignKey.Name, strings.Join(refCols, ", "), f.ForeignKey.ReferencedTableName, f.ForeignKey.ReferencedTableIndex, sb.String())
```

Comment:
The index for the referenced table shouldn't be used here. That position should actually have the referenced columns. To get those though, we'd need access to the referenced table schema, which isn't being passed here. For now, I'd suggest to just back out all of these changes for this function (still use the new `getRowValuesAsString` function though).